pkg/dependency/parser/dart/pub/parse.go

```diff
@@ -16,37 +18,64 @@ const (
 )
 
 // Parser is a parser for pubspec.lock
-type Parser struct{}
+type Parser struct {
+	logger *log.Logger
+}
 
 func NewParser() *Parser {
-	return &Parser{}
+	return &Parser{
+		logger: log.WithPrefix("pub"),
+	}
 }
 
 type lock struct {
-	Packages map[string]Dep `yaml:"packages"`
+	Packages map[string]Dep    `yaml:"packages"`
+	Sdks     map[string]string `yaml:"sdks"`
 }
 
 type Dep struct {
-	Dependency string `yaml:"dependency"`
-	Version    string `yaml:"version"`
+	Dependency  string      `yaml:"dependency"`
+	Version     string      `yaml:"version"`
+	Source      string      `yaml:"source"`
+	Description Description `yaml:"description"`
 }
 
+type Description string
+
 func (p Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependency, error) {
 	l := &lock{}
 	if err := yaml.NewDecoder(r).Decode(&l); err != nil {
 		return nil, nil, xerrors.Errorf("failed to decode pubspec.lock: %w", err)
 	}
 	var pkgs []ftypes.Package
 	for name, dep := range l.Packages {
+		// Some dependencies use one of the SDK versions.
+		// In this case dep.Version == `0.0.0`.
+		// We can't get versions for these dependencies.
+		// Therefore, we use the first version of the SDK constraint specified in the Description.
+		// See https://github.com/aquasecurity/trivy/issues/6017
+		version := dep.Version
+		if version == "0.0.0" && dep.Source == "sdk" {
+			if constraint, ok := l.Sdks[string(dep.Description)]; ok {
+				v, err := firstVersionOfConstrain(constraint)
+				if err != nil {
+					p.logger.Warn("unable to get sdk version from constraint: %w", log.Err(err))
```

Comment:
```suggestion
					p.logger.Warn("Unable to get sdk version from constraint", log.Err(err))
```

Reply:
Fixed in https://github.com/aquasecurity/trivy/pull/6239/commits/191270c0a3fa995e888f1a3d88c29307844d70b4